service/middleware.go

```diff
@@ -118,6 +118,35 @@ func requireUser(onSuccess, onFail http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
+type GqlRequest struct {
+	query string
+}
+
+func (uis *UIServer) noMutationsInProdGQLPlaygroundUnlessSuperUser(next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		reqBody := GqlRequest{}
+		err := utility.ReadJSON(r.Body, &reqBody)
+		if err != nil {
+			uis.LoggedError(w, r, http.StatusBadRequest, errors.New("request has no body"))
+		}
+		referrer := r.Referer()
+		isReferrerPlaygroundInProd := utility.StringSliceContains([]string{referrer}, "evergreen.mongodb.com/graphql")
```

Comment:
is there a way to check if we are in prod by checking against a env variable?

Reply:
We can add environment variables to any environment if needed. We shouldn't hardcode our production URL, as this creates too many possible backdoors, or ways to spoof the referrer to get around this access control.